### PR TITLE
[Doc] 3.1-add-col-mode-partial-update

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 1]
```diff
@@ -2,11 +2,11 @@
 
 Updates rows in a Primary Key table.
 
-In versions earlier than version 3.0, the UPDATE statement only supports single-table UPDATE and does not support common table expressions (CTEs). Starting from version 3.0, StarRocks enriches the syntax to support multi-table joins and CTEs. If you need to join the table to be updated with other tables in the database, you can reference these other tables in the FROM clause or CTE.
+In versions earlier than version 3.0, the UPDATE statement only supports single-table UPDATE and does not support common table expressions (CTEs). Starting from version 3.0, StarRocks enriches the syntax to support multi-table joins and CTEs. If you need to join the table to be updated with other tables in the database, you can reference these other tables in the FROM clause or CTE. Since version 3.1, the UPDATE statement supports the column mode, which are suitable for scenarios involving a small number of columns but a large number of rows, resulting in faster update speeds.
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:5]: readers cannot understand the term “column mode”, change to "column-mode partial update", which is suitable

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 2]
```diff
@@ -57,7 +57,22 @@ One or more other tables in the database. These tables can be joined with the ta
 
 `where_condition`
 
-The condition based on which you want to update rows. Only rows that meet the WHERE condition can be updated. This parameter is required, because it helps prevent you from accidentally updating the entire table. If you want to update the entire table, you can use 'WHERE true'.
+The condition based on which you want to update rows. Only rows that meet the WHERE condition can be updated. This parameter is required, because it helps prevent you from accidentally updating the entire table. If you want to update the entire table, you can use 'WHERE true'. However, this parameter is not mandatory for the [column mode partial updates](#column-mode-partial-updates).
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:60]: "for the" in the last sentence --- delete "the"

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 3]
```diff
@@ -57,7 +57,22 @@ One or more other tables in the database. These tables can be joined with the ta
...
+
+## Column mode partial updates
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:62]: update

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 4]
```diff
@@ -57,7 +57,22 @@ One or more other tables in the database. These tables can be joined with the ta
...
+
+Column mode partial updates are suitable for scenarios where a only small number of columns, but a large number of rows need to be updated. In such scenarios, the column mode offers faster update speeds compared to the regular mode. For example, in a table with 100 columns, if only 10 columns (10% of the total) are updated for all rows, the performance of the column mode is 10 times better than the regular mode.
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:64]: 1. a only small number --- only a small number    2. 英文里不要提 regular mode，中文也可以绕开，就说和没有打开column mode 相比吧，compared to when this mode is not enabled   3. the performance of the column mode is 10 times better than the regular mode. --- the update speed of column mode is 10 times faster.

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 5]
```diff
@@ -57,7 +57,22 @@ One or more other tables in the database. These tables can be joined with the ta
...
+
+The system variable `partial_update_mode` controls the mode of partial updates and supports the following values:
+
+- `auto` (default): The system automatically determines the mode for executing partial updates, which is based on an analysis of the update statement and the columns involved. If the following criteria are met, the system automatically uses the column mode:
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:68]: The system automatically determines the mode of partial updates by analyzing the UPDATE statement and the columns involved. --- based on the analysis 这个名词意思有点弱

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 6]
```diff
@@ -57,7 +57,22 @@ One or more other tables in the database. These tables can be joined with the ta
...
+  - The percentage of updated columns compared to the total number of columns is less than 30%, and the number of updated columns is fewer than 4.
+  - The update statement does not use a WHERE condition.
+Otherwise, the system automatically uses the regular mode.
+
+- `column`: The column mode is used for executing partial updates, which is particularly suitable for partial updates which involve a small number of columns and a large number of rows.
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:73]: The column mode is used for partial updates,

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md [hunk 7]
```diff
@@ -57,7 +57,22 @@ One or more other tables in the database. These tables can be joined with the ta
...
+
+You can use `EXPLAIN UPDATE xxx` to view the mode used for executing partial updates.
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:75]: You can use `EXPLAIN UPDATE xxx` to view the partial update mode.

File: docs/sql-reference/sql-statements/data-manipulation/UPDATE.md (hunk 1 above)
Comment (evelynzhaojie) [reply docs/sql-reference/sql-statements/data-manipulation/UPDATE.md:5]: add privilege requirements for update sql, for example, "This command requires the UPDATE privilege on the table you want to update."

**Review by EsoragotoSpirit — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.1

**Pull request merged**